Makefile

```diff
@@ -1,46 +1,167 @@
 
-tag-images:
+model-api:
+	$(MAVEN) $(ARGS) install -pl parodos-model-api
+
+fast-build-model-api: ARGS = $(FAST_BUILD_ARGS)
+fast-build-model-api: model-api
+
+workflow-engine: ## Build workload engine
+	$(MAVEN) $(ARGS) install -pl workflow-engine
+
+fast-build-workflow-engine: ARGS = $(FAST_BUILD_ARGS) ## Fast build workflow engine
+fast-build-workflow-engine: workflow-engine
+
+pattern-detection-library: ## Build pattern detection library
+	$(MAVEN) $(ARGS) install -pl pattern-detection-library
+
+fast-build-pattern-detection-library: ARGS = $(FAST_BUILD_ARGS) ## Fast build pattern detection library
+fast-build-pattern-detection-library: pattern-detection-library
+
+workflow-examples: ## Build worklow examples
+	$(MAVEN) $(ARGS) -pl workflow-examples
+
+fast-build-workflow-examples: ARGS = $(FAST_BUILD_ARGS) ## Fast build workflow examples
+fast-build-workflow-examples: workflow-examples
+
+coverage: ## Build coverage
+	$(MAVEN) $(ARGS) -pl coverage
+
+fast-build-coverage: ARGS = $(FAST_BUILD_ARGS) ## Fast build coverage
+fast-build-coverage: coverage
+
+# Build docker images
+.PHONY: build-images tag-images push-images push-images-to-kind install-nginx install-kubernetes-dependencies wait-kubernetes-dependencies
+	
+build-images: ## Build docker images
+	$(DOCKER-COMPOSE) -f ./docker-compose/docker-compose.yml build
+
+tag-images: ## Tag docker images with git hash and branch name
 	$(DOCKER) tag docker-compose_workflow-service:latest $(ORG)$(WORKFLOW_SERVICE_IMAGE):$(GIT_HASH)
 	$(DOCKER) tag docker-compose_notification-service:latest $(ORG)$(NOTIFICATION_SERVICE_IMAGE):$(GIT_HASH)
 
 	$(DOCKER) tag docker-compose_workflow-service:latest $(ORG)$(WORKFLOW_SERVICE_IMAGE):$(GIT_BRANCH)
 	$(DOCKER) tag docker-compose_notification-service:latest $(ORG)$(NOTIFICATION_SERVICE_IMAGE):$(GIT_BRANCH)
 
-push-images:
+push-images: ## Push docker images to quay.io registry
 	$(DOCKER) push  $(ORG)/$(WORKFLOW_SERVICE_IMAGE):$(GIT_HASH)
 	$(DOCKER) push  $(ORG)/$(NOTIFICATION_SERVICE_IMAGE):$(GIT_HASH)
 
 	$(DOCKER) push  $(ORG)/$(WORKFLOW_SERVICE_IMAGE):$(GIT_BRANCH)
 	$(DOCKER) push  $(ORG)/$(NOTIFICATION_SERVICE_IMAGE):$(GIT_BRANCH)
 
-push-images-to-kind:
+push-images-to-kind: ## Push docker images to kind
 	$(DOCKER) tag docker-compose_workflow-service:latest $(ORG)$(WORKFLOW_SERVICE_IMAGE):test
 	$(DOCKER) tag docker-compose_notification-service:latest $(ORG)$(NOTIFICATION_SERVICE_IMAGE):test
 	kind load docker-image $(ORG)$(WORKFLOW_SERVICE_IMAGE):test
 	kind load docker-image $(ORG)$(NOTIFICATION_SERVICE_IMAGE):test
 
-install-nginx:
+install-nginx: ## Install nginx
 	kubectl label node kind-control-plane  "ingress-ready=true" || exit 0
 	kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
 
 install-kubernetes-dependencies: install-nginx
 
-wait-kubernetes-dependencies:
+wait-kubernetes-dependencies: ## Wait for ingress to become ready
```

Comment:
I would not use ingress here, mainly because in future we'll need more dependencies in there.